amundsen_application/api/log/v0.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from http import HTTPStatus
+
+from flask import Response, jsonify, make_response, request
+from flask.blueprints import Blueprint
+
+from amundsen_application.log.action_log import action_logging
+from amundsen_application.api.utils.request_utils import get_query_param
+
+
+LOGGER = logging.getLogger(__name__)
+
+log_blueprint = Blueprint('log', __name__, url_prefix='/api/log/v0')
+
+
+@log_blueprint.route('/', methods=['PUT'])
+def log_generic_action() -> Response:
+    """
+    Log a generic action on the frontend. Captured parameters include
+
+    :param command: (Required) Name of action to be logged
+    :param key: (Optional)
+    :param value: (Optional)
+    :param source: (Optional)
+    :return:
+    """
+    @action_logging
+    def _log_generic_action(*, command: str, key: str, value: str, source: str) -> None:
+        pass  # pragma: no cover
+
+    try:
+        command = get_query_param(request.args, 'command', '"command" is a required parameter.')
```

Comment:
Is there a case that we don't need key, value, and source?
If we can make it all required, it would be great to have conventional REST API resource URL `/api/log/v0/command/{command}/source/{source}/...`

Reply:
I'm still trying to figure out the best way to map out the different values that we need to record. There will definitely be cases that each parameter is not required.

For example we want to log the action `"Click on Announcements"`. 

```
{
    command: 'click',
    key: ??,
    value: 'Announcements',
    source: 'navbar',
}
```
Or

```
{
    command: 'navigation',
    label: 'Announcements',
    value: '/announcements', ??
    source: 'click',
}
```
There are are probably 10s of permutations that make sense. Will follow up with PM to figure out the right fields.